tests/unit/test_charm.py

```diff
@@ -300,3 +290,59 @@ def test_get_postgres_password(self, _, __):
         password = self.charm._get_postgres_password()
         self.assertIsNotNone(password)
         self.assertNotEqual(password, "")
+
+    def test_retrieve_resource(self):
+        # Test without adding the resource.
+        path = self.charm._retrieve_resource("key-file")
+        self.assertIsNone(path)
+
+        # Add an empty file as a resource.
+        self.harness.add_resource("key-file", "")
+        # Assert a path is returned.
+        path = self.charm._retrieve_resource("key-file")
+        self.assertTrue(isinstance(path, PosixPath))
+
+    @patch("charm.PostgresqlOperatorCharm._update_config")
+    @patch("ops.model.Container.push")
+    @patch("builtins.open", mock_open(), create=True)
+    def test_store_tls_files(self, _push, _update_config):
+        # Test without uploaded resources.
+        with patch("charm.PostgresqlOperatorCharm._tls_files", PropertyMock(return_value=[])):
+            self.charm._store_tls_files()
+            _push.assert_not_called()
+            _update_config.assert_called_once()
+
+        # Test with only one resource.
+        with patch(
+            "charm.PostgresqlOperatorCharm._tls_files",
+            PropertyMock(return_value=[Path("/tmp/server.key")]),
+        ):
+            self.charm._store_tls_files()
+            _push.assert_not_called()
+            self.assertEqual(_update_config.call_count, 2)
+
+        # Test with both resources.
+        with patch(
+            "charm.PostgresqlOperatorCharm._tls_files",
+            PropertyMock(return_value=[Path("/tmp/server.crt"), Path("/tmp/server.key")]),
+        ):
+            self.charm._store_tls_files()
+            self.assertEqual(_push.call_count, 2)
+            self.assertEqual(_update_config.call_count, 3)
```

Comment:
More of a choice than a comment. 

What do you think about splitting these tests out into 3 separate tests? Same with some of the other units. 
It does add boiler plate, but it might be easier to specify which exact behavior fails during a test failure.
It might also mean we don't have to keep an internal track of `_update_config.call_count` and update that when we modify this test. 

Though it's a nitpick, definitely not important.